### PR TITLE
GH-44954: [C++][CI] Silence protobuf-generated deprecations

### PR DESCRIPTION
**Pull request by pitrou**
### Are there any user-facing changes?

* GitHub Issue: #44954

### CODE REVIEW

**Comment by assignUser** [comment]
:( https://github.com/apache/arrow/actions/runs/12184571186/job/33988782305?pr=44955#step:9:1679

File: cpp/cmake_modules/ThirdpartyToolchain.cmake [hunk 1]
```diff
@@ -2115,7 +2118,9 @@ macro(build_substrait)
   set(SUBSTRAIT_INCLUDES ${SUBSTRAIT_CPP_DIR} ${PROTOBUF_INCLUDE_DIR})
 
   add_library(substrait STATIC ${SUBSTRAIT_SOURCES})
-  set_target_properties(substrait PROPERTIES POSITION_INDEPENDENT_CODE ON)
+  set_target_properties(substrait
+                        PROPERTIES POSITION_INDEPENDENT_CODE ON
+                                   COMPILE_OPTIONS "${SUBSTRAIT_SUPPRESSED_FLAGS}")
```
Comment (kou) [inline cpp/cmake_modules/ThirdpartyToolchain.cmake:2123]: Could you use `target_compile_options()` instead?

We may be able to remove `COMPILE_OPTIONS` source property:
https://github.com/apache/arrow/blob/fe32a7dfe5e22e7737198476fe1ac0e8a5dccef2/cpp/cmake_modules/ThirdpartyToolchain.cmake#L2077-L2078

**Comment by pitrou** [comment]
@github-actions crossbow submit -g cpp

**Comment by pitrou** [comment]
@github-actions crossbow submit -g cpp

**Comment by pitrou** [comment]
AppVeyor failure is unrelated.

**Review by kou — APPROVED** [review]
+1

**Pull request merged**